configure

```diff
@@ -1519,7 +1519,8 @@ fi
 
 nargs=none
 for i in 5 6; do
-  if sh ./trycompile -DSYS_${system} -DNUM_ARGS=${i} gethostbyname.c; then
+  if sh ./trycompile -Werror -DSYS_${system} -DNUM_ARGS=${i} gethostbyname.c
+  then
```

Comment:
What problem are you trying to solve?  Applying a prototyped function to the wrong number of arguments is an error, not a warning.  If you see just a warning, I'm curious to learn what C compiler you're using.

Reply:
If there is a missing prototype, you'll only get warnings.
(observed with gcc)
This was observed in an earlier attempt when using NetBSD/macppc,
while I used system=bsd_elf and not system=netbsd (which the NetBSD/powerpc
portability test now does).

Reply:
Your gcc is strange.
```
$ cat mismatch.c
extern void f(int x, int y);

int main()
{
  f(1);
  f(2,3,4);
  return 0;
}
$ gcc -c mismatch.c
mismatch.c: In function ‘main’:
mismatch.c:5:3: error: too few arguments to function ‘f’
   f(1);
   ^
mismatch.c:1:13: note: declared here
 extern void f(int x, int y);
             ^
mismatch.c:6:3: error: too many arguments to function ‘f’
   f(2,3,4);
   ^
mismatch.c:1:13: note: declared here
 extern void f(int x, int y);
             ^
$ echo $?
1
$ clang -c mismatch.c
mismatch.c:5:6: error: too few arguments to function call, expected 2, have 1
  f(1);
  ~  ^
mismatch.c:1:1: note: 'f' declared here
extern void f(int x, int y);
^
mismatch.c:6:9: error: too many arguments to function call, expected 2, have 3
  f(2,3,4);
  ~     ^
mismatch.c:1:1: note: 'f' declared here
extern void f(int x, int y);
^
2 errors generated.
$ echo $?
1
```

Reply:
Heh, my gcc isn't all that strange, but you didn't test "missing prototype",
you tested "mismatched prototype".
The problem is not what happens with a mismatched prototype.
The problem is what happens where there's a missing prototype:

```
$ cat m.c
int main()
{
  f(1);
  f(2,3,4);
  return 0;
}
$ cc -c m.c
m.c: In function 'main':
m.c:3:3: warning: implicit declaration of function 'f' [-Wimplicit-function-declaration]
   f(1);
   ^
$ echo $?
0
$ cat f.c
void f(int);

void
f(int a)
{
        int x = a * a;
}
$ cc -c f.c
$ echo $?
0
$ cc -o m m.o f.o
$ echo $?
0
$ 
$ cc --version
cc (nb1 20160606) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ 
```

Now...  The reason there is a test for `SYS_netbsd` and  refusal to try to use `gethostbyname_r` there is ... (drum roll...) that the headers on NetBSD do not declare `gethostbyname_r`, but at least at the moment there *is* an implementation of `gethostbyname_r` in libc, so even though you get compile warnings (but not non-zero exit status), you don't get an error during the final linking.

The addition of `-Werror` will avoid any mis-detection of this sort.  At this moment this is therefore a suggested "belts-and-suspenders" safety to avoid this.

It is also likely that the NetBSD library will change to make this not happen...